lib/matplotlib/tests/test_legend.py

```diff
@@ -216,6 +217,158 @@ def test_hatching():
+    ax.legend()
+
+
+@image_comparison(baseline_images=['legend_annotation_text'],
+                  extensions=['png'],
+                  style='mpl20')
+def test_legend_text_annotation():
+    # Related to issue 8236
+    # Tests text annotations showing up in legend
+    fig, ax = plt.subplots(1)
+    ax.set_xticklabels('')
+    ax.set_yticklabels('')
+    my_annotation = ax.annotate("X",
+            xy=(0.1, 0.5),
+            xytext=(0.1, 0.5),
+            color='C2',
+            label='annotation (text, no arrow)')
+    ax.legend(handler_map={my_annotation: HandlerAnnotation(rep_str='Abcde',
+                                                            rep_maxlen=0)})
+
+
+@image_comparison(baseline_images=['legend_annotation_arrow'],
+                  extensions=['png'],
+                  style='mpl20')
+def test_legend_arrow_annotation():
+    # Related to issue 8236
+    # Tests arrow annotations showing up in legend
+    fig, ax = plt.subplots(1)
+    ax.set_xticklabels('')
+    ax.set_yticklabels('')
+    ax.plot([0, 1], [0, 0], label='line1')
+    ax.plot([0, 1], [1, 1], label='line2')
+    ax.annotate("",
+            xy=(0.3, 1.0),
+            xytext=(0.3, 0.0),
+            arrowprops={'arrowstyle': '<->', 'color': 'C7'},
+            label='annotation (no text, arrow)')
+    ax.legend()
+
+
+@image_comparison(baseline_images=['legend_fancy_arrow'],
+                  extensions=['png'],
+                  style='mpl20')
+def test_legend_fancy_arrow_annotation():
+    # Related to issue 8236
+    # Tests arrow annotations showing up in legend
+    fig, ax = plt.subplots(1)
+    ax.set_xticklabels('')
+    ax.set_yticklabels('')
+    arrpatch = mpatches.FancyArrowPatch([0.5, 0.8], [0.9, 0.9],
+                           arrowstyle='<|-',
+                           mutation_scale=20,
+                           color='C3',
+                           label='arrowpatch')
+    ax.add_patch(arrpatch)
+    ax.legend()
+
+
+@image_comparison(baseline_images=['legend_text'],
+                  extensions=['png'],
+                  style='mpl20')
+def test_legend_text():
+    # Related to issue 8236
+    # Tests text showing up in legend
+    fig, ax = plt.subplots(1)
+    ax.set_xticklabels('')
+    ax.set_yticklabels('')
+    ax.text(x=0.1, y=0.1,
+        s='Hello',
+        color='C5',
+        label='text')
+    ax.legend()
+
+
+@image_comparison(baseline_images=['legend_short_text'],
+                  extensions=['png'],
+                  style='mpl20')
+def test_legend_short_text():
+    # Related to issue 8236
+    # Tests short text showing up in legend
+    fig, ax = plt.subplots()
+    ax.set_xticklabels('')
+    ax.set_yticklabels('')
+    ax.text(x=0.1, y=0.2,
+        s='Z',
+        color='C0',
+        label='short text')
+    ax.legend()
+
+
+@image_comparison(baseline_images=['legend_all_annotation_text'],
+                  extensions=['png'],
+                  style='mpl20')
```

Comment:
Why the other tests rather than just this one?

Reply:
Not entirely sure, I'm thinking that he wanted to have them all tested separately and then just a cumulative one, also as a "how to use" kind of thing

Reply:
Image tests have a time cost and a repo-space cost.  I'd remove the redundatn ones unless there is a pressing need for them.

Reply:
got it, push incoming.